topk-py/tests/test_async_collection.py

```diff
@@ -0,0 +1,592 @@
+import asyncio
```

Comment:
the sync and async tests are not testing the same logic twice. for example `test_async_upsert_to_non_existent_collection` has nothing to do with sync/async.

I would just keep minimum number of tests in the async version, such that we test the code is piped properly, but no need to test for behaviour